pkg/scanners/trivy/types.go

```diff
@@ -30,63 +52,97 @@ type (
 	}
 
 	optionSet struct {
-		input string
+		input []string
 		m     map[string]bool
 	}
 
 	ScanStatus int
 
 	Scanner interface {
 		Scan(unversioned.Image) (ScanStatus, error)
+		Timer() *time.Timer
 	}
 )
 
+func DefaultConfig() *Config {
+	return &Config{
+		CacheDir:           "/var/lib/trivy",
+		DBRepo:             "ghcr.io/aquasecurity/trivy-db",
+		DeleteFailedImages: true,
+		Vulnerabilities: VulnConfig{
+			IgnoreUnfixed: true,
+			Types: []string{
+				vulnTypeOs,
+				vulnTypeLibrary,
+			},
+			SecurityChecks: []string{securityCheckVuln},
+			Severities:     []string{severityCritical},
+		},
+		Timeout: TimeoutConfig{
+			Total:    eraserv1.Duration(time.Hour * 23),
+			PerImage: eraserv1.Duration(time.Hour),
+		},
+	}
+}
+
 type ImageScanner struct {
-	scanConfig         scannerSetup
+	trivyScanConfig    scannerSetup
 	imageSourceOptions []fanalImage.Option
+	userConfig         Config
+	timer              *time.Timer
 }
 
 var _ Scanner = &ImageScanner{}
 
+func (s *ImageScanner) Timer() *time.Timer {
+	return s.timer
+}
+
 // Function never returns an error.
 func (s *ImageScanner) Scan(img unversioned.Image) (ScanStatus, error) {
 	refs := make([]string, 0, len(img.Names)+len(img.Digests))
 	refs = append(refs, img.Digests...)
 	refs = append(refs, img.Names...)
 
+	perImageTimeout := time.Duration(s.userConfig.Timeout.PerImage)
+
 	scanSucceeded := false
 	log.Info("scanning image with id", "imageID", img.ImageID, "refs", refs)
 
 	for i := 0; i < len(refs) && !scanSucceeded; i++ {
 		ref := refs[i]
 		log.Info("scanning image with ref", "ref", ref)
 
-		dockerImage, cleanup, err := fanalImage.NewContainerImage(context.Background(), ref, s.scanConfig.dockerOptions, s.imageSourceOptions...)
+		dockerImage, cleanup, err := fanalImage.NewContainerImage(
+			context.TODO(),
```

Comment:
why did we change context here?

Reply:
I think this slipped in from a rebase somewhere. I'll take it out.

Reply:
done